packages/eslint-plugin/tsconfig.json

```diff
@@ -3,7 +3,8 @@
   "compilerOptions": {
     "rootDir": "src",
     "outDir": "dist",
-    "module": "commonjs",
+    "module": "nodenext",
+    "moduleResolution": "nodenext",
```

Comment:
TS ESLint started using export conditions in v6 and stopped supporting module resolutions less than node16 (https://github.com/typescript-eslint/typescript-eslint/issues/7185#issuecomment-1631374643)

Reply:
the `module` option is a bit confusing for me to understand, but from the typescript docs, `nodenext` looks good to me

https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext

does this make a difference to the `@compiled/eslint-plugin` build output? (should we mention it in the changeset?)

and how tricky would it be to update `module` / `moduleResolution` for other packages in Compiled?

Reply:
That's a great question.  It certainly had the potential to, depending on how the build is set up (for instance, I generally use rollup for building packages, which can have a separate set of tsconfig options.  But yeah, I did a before and after comparison of the build output, and the only changes between them are the code changes from this PR (the black text means no difference, red means a difference, and blue is new).  Now, if `target` changed, then that would likely result in more differences, since that determines the specific ECMAScript version that the emitted code should be compatible with.

![image](https://github.com/user-attachments/assets/02928553-3c29-4159-b8a2-fc09284e2722)

As far as what the effort would be to update those props in the other packages: honestly shouldn't be much.  Of those two, `moduleResolution` is the one that will have the most impact on your internal DX, but when you change that, `module` has to come with it.

Reply:
ah cool, yeah that looks fine

no need to do in this PR, but I think it would be good to make the `module` / `moduleResolution` change for the other Compiled packages if practical, just for consistency